server/routes/creatingReports/incidentDetails.ts

```diff
@@ -131,7 +133,17 @@ export default class IncidentDetailsRoutes {
       updatedSection,
       incidentDate ? incidentDate.value : null
     )
-
+    if (incidentDate) {
+      const reportsOnSameDayAsNewReport = await this.reportService.getReportsByDate(
+        parseInt(bookingId, 10),
+        incidentDate.date
+      )
+      req.flash('originalSubmitType', null) // delete any old data
```

Comment:
Flash has been used to record the users intended course following save on incident-details. 
This info is needed to determine the page to display if user selects the 'No' radio in the report-may-already-exist page. 
Flash state for originalSubmitType is nullified to prevent multiple saves, in case user navigates back and forth. 
Testing of flash state in done implicitly in the integration tests

Reply:
the clear out of old state shouldn't be needed? 
could potentially encode this info in the url - by storing it in the flash it won't be resilient to reloading the page.

Reply:
will use query param